[bundler] Add bats tests

james-stocks opened the pull request:
Fixes #2546

Signed-off-by: James Stocks <jstocks@chef.io>

james-stocks commented on bundler/tests/test.sh:
```diff
@@ -0,0 +1,17 @@
+TESTDIR="$(dirname "${0}")"
+
+if [ -z "${1:-}" ]; then
+  echo "Usage: tomcat-native/tests/test.sh FULLY_QUALIFIED_PACKAGE_IDENT"
```
Oops! Updated

predominant commented on bundler/tests/test.bats:
```diff
@@ -0,0 +1,6 @@
+@test "bundler is expected version ${expected_version}" {
+  run hab pkg exec core/bundler bundle -v
+  [ $status -eq 0 ]
+  [[ "$output" =~ "${expected_version}" ]]
```
Why are we doing a regex comparison here?

Also, be wary of the `[[` in bats. They are not required in this context either.

james-stocks replied:
I'll change this to a string comparison, but I used a regex so that our tests are not dependent on the rest of the version string. The version string presented might be reworded arbitrarily or localised.
I'll instead grep the last word from the string, so that we only depend on the version being at the end of the string.

predominant replied:
I'l all for being extremely strict on version checking. I want to know a package is `1.2.3` and not `1.2.3-alpha1`. As such, I use token separation to pull out the exact version string and do a direct string comparison. Theres no confusion or possibility of being shafted by an incorrect version that way.

james-stocks commented:
Changed the version test to be a string equality test, and otherwise updated PR to meet the current testing pattern.

predominant commented:
Based on #2678 lets remove the testing documentation from this README, and consolidate. After that we're good to merge!

james-stocks commented:
Reverted the README and rebased against master so [the tests are passing in this PR](https://buildkite.com/chef-oss/habitat-sh-core-plans-master-verify/builds/1166#eb6cb238-8737-4888-b687-d716039d342b/415-566)

james-stocks commented:
@gavindidrichsen would you mind double checking and merging this please? I think it can merge now but I want to avoid doing an admin merge myself.
